src/components/Buttons/variants.ts

```diff
@@ -87,95 +87,29 @@ export const variants = {
   },
 };
 
-export type ButtonIcon =
-  | 'archive'
-  | 'archivedChats'
-  | 'back'
-  | 'block'
-  | 'blockedChats'
-  | 'close'
-  | 'closeWithBackground'
-  | 'danger'
-  | 'delete'
-  | 'edit'
-  | 'filter'
-  | 'forward'
-  | 'menuDots'
-  | 'menuLines'
-  | 'next'
-  | 'prev'
-  | 'return'
-  | 'rewind'
-  | 'search'
-  | 'searchWithBackground'
-  | 'send'
-  | 'tooltip';
-
 export const iconVariants = {
-  archive: {
-    backgroundImage: `url(${ArchiveIcon})`,
-  },
-  archivedChats: {
-    backgroundImage: `url(${ArchivedChatsIcon})`,
-  },
-  back: {
-    backgroundImage: `url(${BackIcon})`,
-  },
-  block: {
-    backgroundImage: `url(${BlockIcon})`,
-  },
-  blockedChats: {
-    backgroundImage: `url(${BlockedChatsIcon})`,
-  },
-  close: {
-    backgroundImage: `url(${CloseIcon})`,
-  },
-  closeWithBackground: {
-    backgroundImage: `url(${CloseWithBackgroundIcon})`,
-  },
-  danger: {
-    backgroundImage: `url(${DangerIcon})`,
-  },
-  delete: {
-    backgroundImage: `url(${DeleteIcon})`,
-  },
-  edit: {
-    backgroundImage: `url(${EditIcon})`,
-  },
-  filter: {
-    backgroundImage: `url(${FilterIcon})`,
-  },
-  forward: {
-    backgroundImage: `url(${ForwardIcon})`,
-  },
-  menuDots: {
-    backgroundImage: `url(${MenuDotsIcon})`,
-  },
-  menuLines: {
-    backgroundImage: `url(${MenuLinesIcon})`,
-  },
-  next: {
-    backgroundImage: `url(${NextIcon})`,
-  },
-  prev: {
-    backgroundImage: `url(${PrevIcon})`,
-  },
-  return: {
-    backgroundImage: `url(${ReturnIcon})`,
-  },
-  rewind: {
-    backgroundImage: `url(${RewindIcon})`,
-  },
-  search: {
-    backgroundImage: `url(${SearchIcon})`,
-  },
-  searchWithBackground: {
-    backgroundImage: `url(${SearchWithBackgroundIcon})`,
-  },
-  send: {
-    backgroundImage: `url(${SendIcon})`,
-  },
-  tooltip: {
-    backgroundImage: `url(${TooltipIcon})`,
-  },
+  archive: `url(${ArchiveIcon})`,
```

Comment:
Note: This was not part of the PR per say. I did stumble across here when I was implementing the toast-icons, but in the end I had to work differently with it. ( Because I could not change the colors ).

Anyway I refactored this at some point to make it more readable, and inferring the icon-names from the record and not declaring a union by hand.

Reply:
This pattern smells here.


Why no svg icon component + icon prop? No need to change now, just feels weird to me.

Reply:
svg filters or css filters do the magic of changing colors :P

Reply:
yeah, pls share your svg magix